### ISSUE TITLE
Render poster from a given template

### ISSUE THREAD

**Issue by hellozee** [ILUGD/splatter#2]
Currently splatter draws according to a predefined set of hard coded constraints, using templates would able it to create posters with variations.
People could write templates using simple XML, then the simple XML could be compiled into a more complex one which would have things like embedded image files, calculated the constraints for the features.

**Comment by Boot-Error** [comment]
I want to take up this task

### My Idea
Use *yaml* format to design the template, with the semantics similar to css.
Define some elements such as a TextBox, imageBox, and one can write a theme from these primitive elements.
Initially, the theme engine parses the YAML file, determines the elements to be painted. Then the **layout** is calculated, such as overall surface, position of elements on the canvas which then will be supplied to the **painter** which only has code related to painting based on the results from the layout.

I ll start working on theme branch.

**Comment by hellozee** [comment]
Nice, I have never worked with yaml though, can we embed images into the yaml?
Also we have some other important issues to be addressed before starting the theme engine.

**Comment by Boot-Error** [comment]
Um, its more like JSON, I think sticking to JSON files for theme works too, just like sublime themes.

its ok, theme engine can live as a seperate package.

**Comment by hellozee** [comment]
Since there are 2 different files involved and one is already json, making the other file in json would confuse people.
Though the way I perceive it is, a person writes the theme in a standard format like yaml, our theme engine calculates and generates a `.splat` file which will be just an xml but with calculations already done and images embedded into the xml, so that that `.splat` file is now portable and can be used anywhere.
Then anyone could generate poster with that `.splat` and their own `config.json`
